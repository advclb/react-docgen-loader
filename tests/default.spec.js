const compiler = require("./compiler");

test("JSX output match snapshot", async function () {
  const stats = await compiler("prop-types.jsx");
  const output = stats.toJson().modules[0].source;
  expect(output).toMatchSnapshot();
});

test("TSX output match snapshot", async function () {
  const stats = await compiler("typescript.tsx");
  const output = stats.toJson().modules[0].source;
  expect(output).toMatchSnapshot();
});
