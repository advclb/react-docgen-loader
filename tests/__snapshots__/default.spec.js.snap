// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSX output match snapshot 1`] = `"export default {\\"description\\":\\"General component description.\\",\\"displayName\\":\\"MyComponent\\",\\"methods\\":[],\\"props\\":{\\"foo\\":{\\"type\\":{\\"name\\":\\"number\\"},\\"required\\":true,\\"description\\":\\"Description of prop \\\\\\"foo\\\\\\".\\"},\\"bar\\":{\\"type\\":{\\"name\\":\\"custom\\",\\"raw\\":\\"function (props, propName, componentName) {\\\\n  return props + propName + componentName;\\\\n}\\"},\\"required\\":false,\\"description\\":\\"Description of prop \\\\\\"bar\\\\\\" (a custom validation function).\\",\\"defaultValue\\":{\\"value\\":\\"21\\",\\"computed\\":false}},\\"baz\\":{\\"type\\":{\\"name\\":\\"union\\",\\"value\\":[{\\"name\\":\\"number\\"},{\\"name\\":\\"string\\"}]},\\"required\\":false,\\"description\\":\\"\\"}}};"`;

exports[`TSX output match snapshot 1`] = `"export default {\\"description\\":\\"General component description.\\",\\"displayName\\":\\"MyComponent\\",\\"methods\\":[],\\"props\\":{\\"primitive\\":{\\"required\\":true,\\"tsType\\":{\\"name\\":\\"number\\"},\\"description\\":\\"Description of prop \\\\\\"foo\\\\\\".\\"},\\"literalsAndUnion\\":{\\"required\\":true,\\"tsType\\":{\\"name\\":\\"union\\",\\"raw\\":\\"\\\\\\"string\\\\\\" | \\\\\\"otherstring\\\\\\" | number\\",\\"elements\\":[{\\"name\\":\\"literal\\",\\"value\\":\\"\\\\\\"string\\\\\\"\\"},{\\"name\\":\\"literal\\",\\"value\\":\\"\\\\\\"otherstring\\\\\\"\\"},{\\"name\\":\\"number\\"}]},\\"description\\":\\"Description of prop \\\\\\"bar\\\\\\".\\"},\\"arr\\":{\\"required\\":true,\\"tsType\\":{\\"name\\":\\"Array\\",\\"elements\\":[{\\"name\\":\\"any\\"}],\\"raw\\":\\"Array<any>\\"},\\"description\\":\\"\\"},\\"func\\":{\\"required\\":false,\\"tsType\\":{\\"name\\":\\"signature\\",\\"type\\":\\"function\\",\\"raw\\":\\"(value: string) => void\\",\\"signature\\":{\\"arguments\\":[{\\"name\\":\\"value\\",\\"type\\":{\\"name\\":\\"string\\"}}],\\"return\\":{\\"name\\":\\"void\\"}}},\\"description\\":\\"\\"},\\"noParameterName\\":{\\"required\\":false,\\"tsType\\":{\\"name\\":\\"signature\\",\\"type\\":\\"function\\",\\"raw\\":\\"(value: string) => void\\",\\"signature\\":{\\"arguments\\":[{\\"name\\":\\"value\\",\\"type\\":{\\"name\\":\\"string\\"}}],\\"return\\":{\\"name\\":\\"void\\"}}},\\"description\\":\\"\\"},\\"obj\\":{\\"required\\":false,\\"tsType\\":{\\"name\\":\\"signature\\",\\"type\\":\\"object\\",\\"raw\\":\\"{ subvalue?: boolean }\\",\\"signature\\":{\\"properties\\":[{\\"key\\":\\"subvalue\\",\\"value\\":{\\"name\\":\\"boolean\\",\\"required\\":false}}]}},\\"description\\":\\"\\"}}};"`;
